src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -186,7 +187,16 @@ public void trimOldSnapshots() throws Exception {
         trimMap(shardRqMetricsMap, RQ_SNAPSHOTS);
         trimMap(httpRqMetricsMap, HTTP_RQ_SNAPSHOTS);
         trimMap(masterEventMetricsMap, MASTER_EVENT_SNAPSHOTS);
-        trimMap(metricsDBMap, MAX_DATABASES);
+
+        int max_metrics_db_in_memory_objects = PluginSettings.instance().getMaxMetricsDBFilesCount();
```

Comment:
We only allow querying the latest snapshot via the API. Should we fix the max_metrics_db_in_memory_objects to 2 for now? Also, should we manage the db file lifecycle? I propose the following changes - 

* Only manage the latest 2 memory db snapshots. 
* Delete the older snapshots if the setting for db file management is enabled (default).
* Close db connection and dont delete snapshots if db file management is disabled.

What do you think?

Reply:
@ditac 
If I understand you correctly, instead of config `max-metrics-db-files `, we just want a boolean config say `cleanup-old-metricsdb-files` ? If set to 1, we leave out last two files and older metrics-db files and the in-memory snapshots are cleared up. If set to 0, we don't delete the on-disk files but we just cleanup the in memory snapshots leaving behind the two most current ?